### ISSUE TITLE
data set

### ISSUE THREAD

**Issue by yy835055664** [Wadaboa/titanet#3]
Hello, thank you for providing such a good project.
The two data sets librispeech and vctk written in your document, how to create and load personal data sets?

**Comment by Wadaboa** [comment]
Hi there, please have a look at the `src/datasets.py` file. You'd need to create a new subclass of `SpeakerDataset` and implement the 3 abstract methods `get_speakers_utterances`, `get_sample` and `get_path`. You can then use your new custom class in place of `LibriSpeechDataset` and `VCTKDataset`.

Let me know if you have any other questions. Thanks,
Alessio.

**Comment by yy835055664** [comment]
I am very glad that you can provide such careful guidance, and I will continue to reproduce your code according to your ideas.



---Original---
From: "Alessio ***@***.***&gt;
Date: Sun, May 1, 2022 18:33 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Wadaboa/titanet] data set (Issue #3)




 
Hi there, please have a look at the src/datasets.py file. You'd need to create a new subclass of SpeakerDataset and implement the 3 abstract methods get_speakers_utterances, get_sample and get_path. You can then use your new custom class in place of LibriSpeechDataset and VCTKDataset.
 
Let me know if you have any other questions. Thanks,
 Alessio.

**Comment by yy835055664** [comment]
Hi, thank you so much for sharing this code.
Recently, I have been reproducing and learning your project. Regarding custom data sets, both data sets in your project use torchaudio's own data. I have tried custom data in the past few days, but there have been many problems, resulting in failure. Hope you can guide me on non-torchaudio data such as the voxceleb dataset. thank you very much!







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Wadaboa/titanet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月1日(星期天) 晚上6:33
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Wadaboa/titanet] data set (Issue #3)





 
Hi there, please have a look at the src/datasets.py file. You'd need to create a new subclass of SpeakerDataset and implement the 3 abstract methods get_speakers_utterances, get_sample and get_path. You can then use your new custom class in place of LibriSpeechDataset and VCTKDataset.
 
Let me know if you have any other questions. Thanks,
 Alessio.

**Comment by Wadaboa** [comment]
Hi there,

I just pushed a commit to the repo that contains an implementation of VoxCeleb1 and VoxCeleb2. The first version of VoxCeleb is included in the latest release of `torchaudio` (`0.13.0`), hence it was straightforward to implement. Instead, VoxCeleb2 is still not supported by `torchaudio`, so I wrote the implementation from scratch. I haven't tested it throughly, but I hope it can help you move ahead with your experiments. At least it should give you an idea on how to work with non-torchaudio datasets. 

Let me know if you're able to try out the new datasets and, if so, whether you spot any bugs. 

Thanks,
Alessio.
